Libraries/LibWeb/HTML/ElementInternals.h

```diff
@@ -9,10 +9,32 @@
 #include <AK/WeakPtr.h>
 #include <LibGC/Ptr.h>
 #include <LibWeb/Bindings/PlatformObject.h>
+#include <LibWeb/DOM/NodeList.h>
 #include <LibWeb/Forward.h>
+#include <LibWeb/HTML/HTMLElement.h>
+#include <LibWeb/HTML/HTMLFormElement.h>
+#include <LibWeb/HTML/ValidityState.h>
 
 namespace Web::HTML {
 
+struct ValidityStateFlags {
```

Comment:
We already have a `ValdityState` class, it feels like these can be wrapped up a bit more closely.

Reply:
Oh wait, hangon, ignore me, I just realized spec keeps it different with https://github.com/LadybirdBrowser/ladybird/pull/4813/files